packages/editor/src/components/rich-text/tinymce.js

```diff
@@ -337,6 +338,42 @@ export default class TinyMCE extends Component {
 		}
 	}
 
+	/**
+	 * Replaces non breaking spaces inserted by TinyMCE on the space key by
+	 * a normal space.
+	 *
+	 * @param  {SyntheticEvent} event Synthetic key up event.
+	 */
+	onKeyUp( event ) {
+		if ( event.shiftKey || event.keyCode !== SPACE ) {
+			return;
+		}
+
+		const selection = getSelection();
+
+		if ( selection.rangeCount < 0 ) {
+			return;
+		}
+
+		const range = selection.getRangeAt( 0 );
+		const { startContainer, startOffset, collapsed } = range;
+
+		if ( ! collapsed ) {
+			return;
+		}
+
+		const characterOffset = startOffset - 1;
+		const character = startContainer.data[ characterOffset ];
+
+		if ( character !== '\u00a0' ) {
+			return;
+		}
+
+		startContainer.deleteData( characterOffset, 1 );
```

Comment:
Any reason to delete + insert vs. `replaceData( characterData, 1, ' ' )` ?

Reply:
I initially tried that. We could do that but also have to set the selection back manually. Opted for the browser doing it to avoid mistakes.